### PR TITLE
Implemented class for theme  

### PR DESCRIPTION
**Pull request by narensingh008**

### CODE REVIEW

File: DesktopModules/Vanjaro/Core/Library/Components/Theme.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        #endregion
+
+        public Theme()
+        {
+        }
+        public Theme(int PortalID)
+        {
+            this.PortalID = PortalID;
+        }
+        public string ThemeName
```
Comment (meetmandeep) [inline DesktopModules/Vanjaro/Core/Library/Components/Theme.cs]: ThemeName should be just "Name"

File: DesktopModules/Vanjaro/Core/Library/Components/Theme.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                return ThemeValue;
+            }
+        }
+
+        public string EditLayout
+        {
+            get
+            {
+                string FolderPath = HttpContext.Current.Server.MapPath("~/Portals/_default/vThemes/" + ThemeName + "/Layout.Edit.html");
+                return System.IO.File.ReadAllText(FolderPath);
```
Comment (meetmandeep) [inline DesktopModules/Vanjaro/Core/Library/Components/Theme.cs]: I don't see any cache here or on the theme object.

File: DesktopModules/Vanjaro/Core/Library/Managers/ThemeManager.cs [hunk 3]
```diff
@@ -21,21 +21,21 @@ public static partial class Managers
     {
         public class ThemeManager
         {
-            public static string GetCurrentThemeName(int? PortalID = null)
+            public static Theme GetCurrent(int? PortalID = null)
```
Comment (meetmandeep) [inline DesktopModules/Vanjaro/Core/Library/Managers/ThemeManager.cs]: This should not be a method but a singleton. For example, ThemeManager.CurrentTheme.Name

**Review by meetmandeep — CHANGES_REQUESTED** [review]

File: DesktopModules/Vanjaro/Core/Library/Components/Theme.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using static Vanjaro.Core.Managers;
+
+namespace Vanjaro.Core.Components
+{
+
+    public class Theme
```
Comment (meetmandeep) [inline DesktopModules/Vanjaro/Core/Library/Components/Theme.cs:11]: Where are other objects such as Guid?

File: DesktopModules/Vanjaro/Core/Library/Components/Theme.cs (hunk 1 above)
Comment (narensingh008) [reply DesktopModules/Vanjaro/Core/Library/Components/Theme.cs]: Fixed

File: DesktopModules/Vanjaro/Core/Library/Components/Theme.cs (hunk 2 above)
Comment (narensingh008) [reply DesktopModules/Vanjaro/Core/Library/Components/Theme.cs]: Fixed

File: DesktopModules/Vanjaro/Core/Library/Managers/ThemeManager.cs (hunk 3 above)
Comment (narensingh008) [reply DesktopModules/Vanjaro/Core/Library/Managers/ThemeManager.cs]: We require a portalid when the calls from IUpgradeable.

Comment (meetmandeep) [reply DesktopModules/Vanjaro/Core/Library/Managers/ThemeManager.cs]: @narensingh008 @mksmanish79 See mee to discuss these changes.

**Review by meetmandeep — CHANGES_REQUESTED** [review]

**Review by meetmandeep — APPROVED** [review]

**Pull request merged**